encryptio.py

```diff
@@ -1,55 +1,302 @@
+            return
+
+    cipher = Cipher(algorithms.IDEA(key), mode=mode)
+    encryptor = cipher.encryptor()
+    ciphertext = encryptor.update(data) + encryptor.finalize()
+    create_binary_file(ciphertext, 'IDEA0000', [])
+
+
+def encrypt_blowfish(data, key, mode):
+    # py encryptio.py encryption BLOWFISH 'variablekeyleange' CBC 'teststring'
+    match mode:
+        case CryptoModes.CBC:
+            mode = Blowfish.MODE_CBC
+            data = pad(data, Blowfish.block_size)
+        case CryptoModes.CFB:
+            mode = Blowfish.MODE_CFB
+        case CryptoModes.CTR:
+            # TODO: discuss problem: https://stackoverflow.com/questions/52787147/using-ctr-mode-in-des-algorithm-in-python
+            mode = Blowfish.MODE_CTR
+            cipher = Blowfish.new(key, mode, nonce=b'')
+        case CryptoModes.EAX:
+            mode = Blowfish.MODE_EAX
+        case CryptoModes.ECB:
+            mode = Blowfish.MODE_ECB
+            data = pad(data, Blowfish.block_size)
+        case CryptoModes.OFB:
+            mode = Blowfish.MODE_OFB
+        case _:
+            print('This mode is not available for the selected algorithm')
+            return
+
+    if mode != Blowfish.MODE_CTR:
+        cipher = Blowfish.new(key, mode)
+
+    if mode == Blowfish.MODE_EAX:
+        ciphertext, tag = cipher.encrypt_and_digest(data)
+    else:
+        ciphertext = cipher.encrypt(data)
+
+    if mode == Blowfish.MODE_EAX:
+        additional_information = ['nonce_tag', cipher.nonce + tag]
+    elif mode == AES.MODE_ECB:
+        additional_information = []
+    else:
+        additional_information = ['initialization_vector', cipher.iv]
+
+    create_binary_file(ciphertext, 'BLOWFISH', additional_information)
+
+
+## mode execution
+def encrypt_mode(args):
+    match args.cryptoAlgorithm:
+        case CryptoAlgorithms.AES:
+            encrypt_aes(str.encode(args.cryptoData), str.encode(args.cryptoKey), args.cryptoMode)
+        case CryptoAlgorithms.DES:
+            encrypt_des(str.encode(args.cryptoData), str.encode(args.cryptoKey), args.cryptoMode, DES)
+        case CryptoAlgorithms.TRIPLEDES:
+            encrypt_des(str.encode(args.cryptoData), str.encode(args.cryptoKey), args.cryptoMode, DES3)
+        case CryptoAlgorithms.IDEA:
+            encrypt_idea(str.encode(args.cryptoData), str.encode(args.cryptoKey), args.cryptoMode)
+        case CryptoAlgorithms.BLOWFISH:
+            encrypt_blowfish(str.encode(args.cryptoData), str.encode(args.cryptoKey), args.cryptoMode)
+
+
+# def decrypt_mode(args):
+
+# def bruteforce_mode(args):
+
+## mode listing
+def arg_encryption_mode(args):
+    encrypt_mode(args)
+
+
+def arg_decryption_mode(args):
+    encrypt_mode(args)
+
+
+# def arg_bruteforce_mode(args):
+#     bruteforce_mode(args)
+
+
 ## argument checker
 def check_key(key):
-
     if not (key.isalpha()):
         raise argparse.ArgumentTypeError('Key is not a string')
 
-    if(len(key) < 8): 
+    if len(key) < 8:
```

Comment:
~~The key length should be adjusted according to the encryption algorithms. 
(64, 128, 256, ..)~~
The key length for the individual encryption algorithms should be checked. 
@tzigiannis  you suggested to add this to the enum of CryptoAlgorithms, do you know if this would work? That would be a great way.

Reply:
Argparser can only pass one value to callable in type. I did the check in the arg_encryption_mode method.